File: src/dataset/Java/armstrong-numbers_exercism-awsaf174/claude/ClaudeSonnet4_ArmstrongNumbers_v2.java
class ArmstrongNumbers {

    boolean isArmstrongNumber(int numberToCheck) {
        if (numberToCheck < 0) return false;
        if (numberToCheck < 10) return true;
        
        int temp = numberToCheck;
        int digits = 0;
        
        // Count digits
        while (temp > 0) {
            temp /= 10;
            digits++;
        }
        
        temp = numberToCheck;
        long sum = 0;
        
        // Calculate sum of powers
        while (temp > 0) {
            int digit = temp % 10;
            sum += pow(digit, digits);
            temp /= 10;
        }
        
        return sum == numberToCheck;
    }
    
    private long pow(int base, int exp) {
        long result = 1;
        for (int i = 0; i < exp; i++) {
            result *= base;
        }
        return result;
    }
}

File: src/dataset/c/armstrong-numbers_exercism-viettronics868/claude/ClaudeSonnet4_armstrong_numbers_v2.c
#include "armstrong_numbers.h"
#include <stdint.h>
#include <stdio.h>
#include <stdbool.h>
#include <stddef.h>

static int int_pow(int base, int exp) {
    int result = 1;
    while (exp > 0) {
        if (exp & 1) {
            result *= base;
        }
        base *= base;
        exp >>= 1;
    }
    return result;
}

bool is_armstrong_number(int candidate) {
    if (candidate < 0) return false;
    if (candidate < 10) return true;
    
    int temp = candidate;
    int digits = 0;
    
    // Count digits
    while (temp > 0) {
        digits++;
        temp /= 10;
    }
    
    temp = candidate;
    int sum = 0;
    
    // Calculate sum of powers
    while (temp > 0) {
        int digit = temp % 10;
        sum += int_pow(digit, digits);
        temp /= 10;
    }
    
    return sum == candidate;
}